src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -30,11 +31,28 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// <returns>A driver that works with the board the program is executing on.</returns>
         private static GpioDriver GetBestDriverForBoard()
         {
-            string[] cpuInfoLines = File.ReadAllLines(CpuInfoPath);
-            Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
+            string[] cpuInfoLines;
+            Regex regex;
+
+            // a first chance of finding a raspberry pi..... perhaps not on Raspbian. note
+            // that we want this to fail quietly if something goes wrong with the test and continue
+            // with detecting a PI usiing /dev/cpuinfo
+            try
+            {
+                if (File.ReadAllText(ModelInfoPath).StartsWith("Raspberry Pi"))
+                {
+                    return new RaspberryPi3Driver();
+                }
+            }
+            catch { }
```

Comment:
when do we expect exceptions here? If this is just reading the file should you instead just call `File.Exists`?

Reply:
@krwq my thoughts behind this is that I was trying make it as simple as possible I.e. just a single line to do the testing. You would probably want to ignore errors at this point because you are not sure how future operating Linux distributions will do things so I didn’t see why catching the non existence of a file with an exception would be an issue in this case.

Reply:
Non rpi boards may not have the file or rpi distributions that do not use a variant of the standard device tree and firmware.

Reply:
actually I think it might make sense to let it throw for new board - this way we can react early to any differences on the new distros - if we choke the error we won't know about it until it bites us